pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_PythonPrint(self, expr):
+        end = '\n'
+        sep = ' '
+        code = ''
+        empty_end = FunctionCallArgument(LiteralString(''), 'end')
+        space_end = FunctionCallArgument(LiteralString(' '), 'end')
+        for f in expr.expr:
+            if f.has_keyword:
+                if f.keyword == 'sep':
+                    sep = str(f.value)
+                elif f.keyword == 'end':
+                    end = str(f.value)
+        args_format = []
         args = []
-        n = len(expr.expr)
-        for j, f in enumerate(expr.expr):
+        orig_args = [f for f in expr.expr if not f.has_keyword]
+
+        def formatted_args_to_print(fargs_format, fargs, end):
+
+            def split_long_argument(line):
+                line = line[1 : len(line) - 1]
+                length = len(line)
+                longest_chunk = length
+                cuts = []
+                positions = [idx for idx in range(length) if line[idx] == ' ']
+                if len(positions) > 12:
+                    positions = positions[int(len(positions) / 4) : 12]
+                for mask in range(1, 1<<len(positions)):
+                    mask_cuts = [positions[idx] for idx in range(len(positions)) if (1 << idx)&mask != 0 ]
+                    mask_cuts.append(length)
+                    valid = True
+                    mask_longest_chunk = 0
+                    for idx in range(len(mask_cuts) - 1):
+                        if mask_cuts[idx + 1] - mask_cuts[idx] < 8:
+                            valid = False
+                        mask_longest_chunk = max(mask_longest_chunk, mask_cuts[idx + 1] - mask_cuts[idx])
+                    if valid and mask_longest_chunk < longest_chunk:
+                        cuts = mask_cuts
+                        longest_chunk = mask_longest_chunk
```

Comment:
Why do we need all the chunks to have at least 8 characters?

Reply:
because we don't want too small of a chunk. Minimizing the longest chunk will favor more cuts, thus the optimal combination will have as many cuts as possible, practically a cut at every white space. But we don't want that, that's why i limited the distance between two cuts.

Reply:
Ah ok. Yes splitting at every space would be excessive